File: android2_lesson01/src/java/java/com/example/ama/android2_lesson01/ui/details/mvp/base/DetailsPresenter.java
package com.example.ama.android2_lesson01.ui.details.mvp.base;

import com.example.ama.android2_lesson01.model.Note;
import com.example.ama.android2_lesson01.ui.base.Presenter;

public interface DetailsPresenter<T extends DetailsNoteView> extends Presenter<T> {
    void createNote(String title, String text);

    void updateNote(Note targetNote, String newTitle, String newText);
}

File: android2_lesson01/src/kotlin/java/com/example/ama/android2_lesson01/ui/details/mvp/base/DetailsPresenter.kt
package com.example.ama.android2_lesson01.ui.details.mvp.base

import com.example.ama.android2_lesson01.model.Note
import com.example.ama.android2_lesson01.ui.base.Presenter

interface DetailsPresenter<T : DetailsNoteView> : Presenter<T> {
    fun createNote(title: String, text: String)
    fun updateNote(targetNote: Note, newTitle: String, newText: String)
}